pkg/template/controller/templateinstance_controller.go

```diff
@@ -2,37 +2,37 @@ package controller
 
 import (
 	"fmt"
-	"net/http"
 
-	"github.com/openshift/origin/pkg/api/latest"
-	authclient "github.com/openshift/origin/pkg/auth/client"
-	"github.com/openshift/origin/pkg/client"
-	"github.com/openshift/origin/pkg/config/cmd"
-	templateapi "github.com/openshift/origin/pkg/template/api"
-	templateapiv1 "github.com/openshift/origin/pkg/template/api/v1"
-	templateclientset "github.com/openshift/origin/pkg/template/clientset/internalclientset"
 	kapi "k8s.io/kubernetes/pkg/api"
 	"k8s.io/kubernetes/pkg/api/meta"
 	"k8s.io/kubernetes/pkg/api/meta/metatypes"
 	"k8s.io/kubernetes/pkg/api/unversioned"
 	"k8s.io/kubernetes/pkg/auth/user"
 	"k8s.io/kubernetes/pkg/client/cache"
-	kclientset "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
 	"k8s.io/kubernetes/pkg/client/restclient"
 	"k8s.io/kubernetes/pkg/kubectl/resource"
 	"k8s.io/kubernetes/pkg/runtime"
 	utilruntime "k8s.io/kubernetes/pkg/util/runtime"
 	"k8s.io/kubernetes/pkg/watch"
+
+	"github.com/openshift/origin/pkg/api/latest"
+	authclient "github.com/openshift/origin/pkg/auth/client"
+	"github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/config/cmd"
+	templateapi "github.com/openshift/origin/pkg/template/api"
+	templateapiv1 "github.com/openshift/origin/pkg/template/api/v1"
+	templateclientset "github.com/openshift/origin/pkg/template/clientset/internalclientset"
+	internalversiontemplate "github.com/openshift/origin/pkg/template/clientset/internalclientset/typed/template/internalversion"
 )
 
 type TemplateInstanceController struct {
-	restconfig     *restclient.Config
-	templateclient *templateclientset.Clientset
-	controller     *cache.Controller
+	restconfig     restclient.Config
+	templateclient internalversiontemplate.TemplateInterface
+	controller     cache.ControllerInterface
 }
 
-func NewTemplateInstanceController(restconfig *restclient.Config) *TemplateInstanceController {
-	c := TemplateInstanceController{restconfig: restconfig, templateclient: templateclientset.NewForConfigOrDie(restconfig)}
+func NewTemplateInstanceController(restconfig restclient.Config) *TemplateInstanceController {
```

Comment:
(and because doing a shallow copy gives you a false sense of security where you can still mutate parts of the original object.  at least when you take a pointer you know you need to be careful)